### PR TITLE
Adds NUM_TIMES_SAMPLED agent metadata to the submitted profile

### PR DESCRIPTION
**Pull request by ysshao96**
*Description of changes:*
Adds serialization logic to submit `total_sample_count` as  NUM_TIMES_SAMPLED property of the AgentMetadata.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by pandpara — COMMENTED** [review]
Can you update the minimum samples in a profile to 1 to make it consistent with java agent : https://github.com/aws/amazon-codeguru-profiler-python-agent/blob/main/codeguru_profiler_agent/profiler_disabler.py#L9

File: codeguru_profiler_agent/profiler_disabler.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 logger = logging.getLogger(__name__)
 CHECK_KILLSWITCH_FILE_INTERVAL_SECONDS = 60
 MINIMUM_MEASURES_IN_DURATION_METRICS = 20
-MINIMUM_SAMPLES_IN_PROFILE = 5
+MINIMUM_SAMPLES_IN_PROFILE = 1
```
Comment (mirelap-amazon) [inline codeguru_profiler_agent/profiler_disabler.py:9]: I know you got this as a comment from @pandpara , but I don't think this should be updated, and especially in this commit. This affects if the profiler is disabled or not and it's different than what the general message of this commit is about. If there's a concern about this value, we can do it in a different commit (even release).

**Review by mirelap-amazon — CHANGES_REQUESTED** [review]
One comment about the MINIMUM_SAMPLES_IN_PROFILE

**Review by mirelap-amazon — APPROVED** [review]

**Comment by Shay-Jin** [comment]
Hi Max, could you have a `numTimesSampled > 0` check before submitting profiles?

**Pull request merged**